aws/image/misc-settings.sh

```diff
@@ -20,24 +20,105 @@ if [ "$CLOUD_PROVIDER" == "vsphere" ]
 then
 # Add vsphere specific commands to execute on remote
     case $PODVM_DISTRO in
-	centos)
-	    #fallthrough
-	    ;&
-	rhel)
-	    (! dnf list --installed | grep open-vm-tools >/dev/null 2>&1) && \
-		(! dnf -y install open-vm-tools) && \
-		     echo "$PODVM_DISTRO: Error installing package required for cloud provider: $CLOUD_PROVIDER" 1>&2 && exit 1
-	    ;;
-	ubuntu)
-	    (! dpkg -l | grep open-vm-tools > /dev/null 2>&1) && apt-get update && \
-		(! apt-get -y install open-vm-tools > /dev/null 2>&1) && \
-		     echo "$PODVM_DISTRO: Error installing package required for cloud provider: $CLOUD_PROVIDER" 1>&2  && exit 1
-	    ;;
-	*)
-	    ;;
+    centos)
+        #fallthrough
+        ;&
+    rhel)
+        (! dnf list --installed | grep open-vm-tools > /dev/null 2>&1) && \
+        (! dnf -y install open-vm-tools) && \
+             echo "$PODVM_DISTRO: Error installing package required for cloud provider: $CLOUD_PROVIDER" 1>&2 && exit 1
+        ;;
+    ubuntu)
+        (! dpkg -l | grep open-vm-tools > /dev/null 2>&1) && apt-get update && \
```

Comment:
Ideally should be handled at the packer/image-build time. But I think we don't have a way of saying if we need such tunneling on all VM types? I am guessing since we have identified that we need it on Azure, we can make it default?

Reply:
The only reason to have it like this is to use the same file across podvm/aws/azure directory. Eventually the plan is to remove these duplicates - https://github.com/confidential-containers/cloud-api-adaptor/issues/899. It's unmaintainable otherwise.